### PR TITLE
escape branch names in the query as well

### PR DESCRIPTION
**Pull request by krzyzacy**
branch can also have slashes in the name, aka, branch: foo/bar/baz is valid, but need to be escaped when calling the gerrit rest api.

### CODE REVIEW

**Comment by andygrunwald** [comment]
@krzyzacy Thank you. WOuld you mind to add also a basic test case for this? To avoid a regression in the future?

**Comment by krzyzacy** [comment]
@andygrunwald will do :+1:

**Comment by andygrunwald** [comment]
@krzyzacy Thank you. Once a simple test case with and without escaping is there, I will merge it directly.

File: projects_test.go [hunk 1]
```diff
@@ -192,3 +192,75 @@ func ExampleProjectsService_ListProjects() {
...
+			Ref:       "456",
+			Revision:  "deadbeef",
+			CanDelete: false,
+		},
+	}
+
+	testMux.HandleFunc("/projects/go/branches/", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+
+		branchName := r.URL.Path[len("/projects/go/branches/"):]
```
Comment (krzyzacy) [inline projects_test.go]: (note that the branchName is already transformed back to unescaped slashes so it didn't actually provide actual value to what's been changed in the current PR :crying_cat_face: )

Comment (andygrunwald) [reply projects_test.go]: Seems to be a Gerrit vs. Test behavior here. Thanks anyway!

**Comment by andygrunwald** [comment]
Thank you!

**Pull request merged**